airbyte-scheduler/src/main/java/io/airbyte/scheduler/client/CachingSchedulerJobClient.java

```diff
@@ -22,36 +22,14 @@
  * SOFTWARE.
  */
 
-package io.airbyte.server.cache;
+package io.airbyte.scheduler.client;
 
-import io.airbyte.protocol.models.ConnectorSpecification;
-import java.util.Optional;
-
-public interface SpecCache {
-
-  Optional<ConnectorSpecification> get(String imageName);
-
-  void put(String imageName, ConnectorSpecification spec);
-
-  void evict(String imageName);
-
-  public static class AlwaysMissCache implements SpecCache {
-
-    @Override
-    public Optional<ConnectorSpecification> get(String imageName) {
-      return Optional.empty();
-    }
-
-    @Override
-    public void put(String imageName, ConnectorSpecification spec) {
-      // no op.
-    }
-
-    @Override
-    public void evict(String imageName) {
-      // no op.
-    }
+/**
+ * This client is meant to be an interface over a cached implementation of
+ * {@link SchedulerJobClient}. It exposes functionality to allow invalidating the cache.
+ */
+public interface CachingSchedulerJobClient extends SchedulerJobClient {
 
-  }
+  void resetCache();
```

Comment:
this is the part i mentioned in the PR description. caching isn't fully hidden. but couldn't really think of another way to allow the caller to invalidate the cache without exposing it like this.